TYPO3.Neos/Classes/TYPO3/Neos/Routing/Cache/RouteCacheFlusher.php

```diff
@@ -52,6 +53,24 @@ public function registerNodeChange(NodeInterface $node)
     }
 
     /**
+     * Schedules flushing of the all routing cache entries of the workspace whose base workspace has changed.
+     * In most cases $workspace will be a user's personal workspace. Flushing the respective cache entries guards
+     * against mismatches for nodes which exist in the old and the new base workspace but have different node
+     * identifiers and the same URI path (segment).
+     *
+     * @param Workspace $workspace
+     * @param Workspace|null $oldBaseWorkspace
+     * @param Workspace|null $newBaseWorkspace
+     * @return void
+     */
+    public function registerBaseWorkspaceChange(Workspace $workspace, Workspace $oldBaseWorkspace = null, Workspace $newBaseWorkspace = null)
```

Comment:
These arguments are provided by the respective emit method - I don't know what's more messy, to not use them in the slot or to have a method signature which does not comply with "the interface" of the signal.

I don't mind removing the arguments, but I think it's not better or worse this or the other way.